examples/gno.land/r/demo/realm_banker/realm_banker.gno

```diff
@@ -0,0 +1,39 @@
+package realm_banker
+
+import (
+	"std"
+)
+
+var denoms map[string]std.Address = make(map[string]std.Address)
+
+func Mint(addr std.Address, denom string, amount int64) {
+	caller := std.GetOrigCaller()
+
+	minter, ok := denoms[denom]
+	if ok {
+		if minter != caller {
+			panic("not minter")
+		}
+	} else {
+		denoms[denom] = caller
+	}
+
+	issuer := std.GetBanker(std.BankerTypeRealmIssue)
+	issuer.IssueCoin(addr, denom, amount)
+}
+
+func Burn(addr std.Address, denom string, amount int64) {
+	caller := std.GetOrigCaller()
+
+	minter, ok := denoms[denom]
+	if ok {
+		if minter != caller {
+			panic("not minter")
+		}
+	} else {
+		denoms[denom] = caller
+	}
```

Comment:
Nitpick: you can simplify this as
```go
if ok && minter != caller {
    panic("not minter")
}

denoms[denom] = caller
// ...
```

Reply:
Resolved in https://github.com/gnolang/gno/pull/875/commits/31880990e97fb81035aad8f861d06193981b885f